lambdas/requirements/layers/requirements_core_lambda_layer.txt

```diff
@@ -17,8 +17,8 @@ pypdf==3.17.2
 python-dateutil==2.8.2
 requests==2.32.0
 responses==0.23.1
-s3transfer==0.8.2
+s3transfer==0.10.1
 setuptools==65.5.1
 six==1.16.0
 types-PyYAML==6.0.12.11
-urllib3==2.0.7
+urllib3==2.2.2
```

Comment:
I bumped up the version for urllib3 as pipeline audit detect a recent vuln issue:
https://github.com/nhsconnect/national-document-repository/actions/runs/9564294917/job/26364682299

also the version boto3 / botocore / s3transfer was bumped up, as botocore internally use urllib3 as well